### PR TITLE
Not fail the promise when a closed Channel is offered back to the Cha…

### PR DESCRIPTION
**Pull request by normanmaurer**
…nnelPool

Motivation:

We should not fail the promise when a closed Channel is offereed back to the ChannelPool as we explicit mention that the Channel must always be returned.

Modifications:

- Not fail the promise
- Add test-case

Result:

Fixes [#6831]

### CODE REVIEW

**Comment by normanmaurer** [comment]
PTAL @fian46

File: transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java [hunk 1]
```diff
@@ -327,7 +324,7 @@ private void releaseAndOfferIfHealthy(Channel channel, Promise<Void> promise, Fu
             releaseAndOffer(channel, promise);
         } else { //channel ont healthy, just releasing it.
```
Comment (Scottmitch) [inline transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java]: `ont` -> `not`

File: transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java [hunk 2]
```diff
@@ -300,6 +300,35 @@ public void run() {
         channel.close().syncUninterruptibly();
     }
 
+    @Test
+    public void testReleaseClosed() {
+        EventLoopGroup group = new LocalEventLoopGroup(1);
```
Comment (Scottmitch) [inline transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java]: shouldn't we be closing this at the end of the test? consider creating this in a `@BeforeClass` and cleaning up in `@AfterClass`.

Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java]: yes! Let me update

File: transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java (hunk 1 above)
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java]: +1

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (94c0ef3c96189b3fb02ee68a4e229d1e3ba6e2df) and 4.0 (8b66dc7c16)

**Comment by fian46** [comment]
nice already fixed and closed

**Pull request closed without merging**